src/security/Crypto.js

```diff
@@ -1,135 +1,183 @@
-import nacl from "tweetnacl";
-import sha256 from "fast-sha256";
 import base64Url from "base64url";
 import stringify from "json-stable-stringify";
-import Util from "../Util";
 import {Buffer} from "buffer/.";
+import nacl from "tweetnacl";
+import sha256 from "fast-sha256";
 
-let sjcl = null;
-
-/**
- * Initializes sjcl and mouse input collection, for IE10
- */
-if (!BROWSER) {
-} else if (window.crypto && window.crypto.getRandomValues) {
-  // Do nothing, we have secure random crypto
-} else if (window.msCrypto && window.msCrypto.getRandomValues) {
-  // Do nothingm, we have secure random crypto
-} else {
-    // Only set it up when it is necessary
-    sjcl = require('sjcl');
-    sjcl.random.startCollectors();
+if (BROWSER && !crypto) {
+    throw new Error('Your browser does not support Web Cryptography API');
 }
 
 /**
  * Class providing static crypto primitives.
  */
 class Crypto {
```

Comment:
Instead of doing if(BROWSER) for everything, can we have two cryptoengine classes: one that uses libsodium and another that uses webcrypto? and push common methods to the cryptoengine

Reply:
we can but i think I'm going to refactor this in another PR.